src/main.rs

```
@@ -209,75 +204,64 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
     }
 
     if args.cdawg {
-        return match args.disk_path.clone() {
-            Some(path) => {
-                if args.ram {
-                    println!("Building CDAWG in RAM but saving on disk...");
-                    type Mb = RamBacking<N, (DefaultIx, DefaultIx), DefaultIx>;
-                    let mb = Mb::default();
-                    return Ok(build_cdawg::<Mb>(args, mb)?);
-                }
-                println!("Building CDAWG on disk...");
-                type Mb = DiskBacking<N, (DefaultIx, DefaultIx), DefaultIx>;
-                let mb = Mb::new(path);
-                Ok(build_cdawg::<Mb>(args, mb)?)
-            }
-            None => {
-                println!("Building CDAWG in RAM...");
+        return if !args.save_path.is_empty() {
```

The logic here looks wrong in that the outer `else` should raise an error rather than duplicating the RAM logic. In other words, we probably want to require that `args.save_path` since even if we build in RAM, we will end up saving there eventually.

Options we want to support:
1. Build in RAM, save to disk (if `--ram` is set)
2. Build in disk, save to disk

---

Is there never a case where people just want to use the dawg in ram without saving?

---

In that case you can pass /dev/null as the path.

Alternatively, you could make it so that's what happens when save_path isn't set. But probably there should be a warning or something